### ISSUE TITLE
background acc to be 0

### ISSUE THREAD

**Issue by LansonChen** [open-mmlab/mmsegmentation#1676]
![image](https://user-images.githubusercontent.com/33587434/174101890-72827854-9b89-42b4-85cc-82c7573bce47.png)
here's my config in jupyter:
**Config:**
norm_cfg = dict(type='BN', requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained='open-mmlab://resnet50_v1c',
    backbone=dict(
        type='ResNetV1c',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        dilations=(1, 1, 2, 4),
        strides=(1, 2, 1, 1),
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        contract_dilation=True),
    decode_head=dict(
        type='DAHead',
        in_channels=2048,
        in_index=3,
        channels=512,
        pam_channels=64,
        dropout_ratio=0.1,
        num_classes=2,
        norm_cfg=dict(type='BN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    auxiliary_head=dict(
        type='FCNHead',
        in_channels=1024,
        in_index=2,
        channels=256,
        num_convs=1,
        concat_input=False,
        dropout_ratio=0.1,
        num_classes=2,
        norm_cfg=dict(type='BN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4)),
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))
dataset_type = 'MBdDataset'
data_root = 'building'
img_norm_cfg = dict(
    mean=[77.434, 79.858, 70.154], std=[48.68, 45.53, 47.354], to_rgb=True)
crop_size = (256, 256)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=False),
    dict(type='Resize', img_scale=(256, 256), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=(256, 256), cat_max_ratio=0.75),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(
        type='Normalize',
        mean=[77.434, 79.858, 70.154],
        std=[48.68, 45.53, 47.354],
        to_rgb=True),
    dict(type='Pad', size=(256, 256), pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(256, 256),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[77.434, 79.858, 70.154],
                std=[48.68, 45.53, 47.354],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=8,
    workers_per_gpu=8,
    train=dict(
        type='MBdDataset',
        data_root='building',
        img_dir='images/training',
        ann_dir='annotations/training',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', reduce_zero_label=False),
            dict(type='Resize', img_scale=(256, 256), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(256, 256), cat_max_ratio=0.75),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[77.434, 79.858, 70.154],
                std=[48.68, 45.53, 47.354],
                to_rgb=True),
            dict(type='Pad', size=(256, 256), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='MBdDataset',
        data_root='building',
        img_dir='images/validation',
        ann_dir='annotations/validation',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(256, 256),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[77.434, 79.858, 70.154],
                        std=[48.68, 45.53, 47.354],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='MBdDataset',
        data_root='building',
        img_dir='images/test',
        ann_dir='annotations/test',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(256, 256),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[77.434, 79.858, 70.154],
                        std=[48.68, 45.53, 47.354],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
log_config = dict(
    interval=10, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
cudnn_benchmark = True
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0005)
optimizer_config = dict()
lr_config = dict(policy='poly', power=0.9, min_lr=0.0001, by_epoch=False)
runner = dict(type='IterBasedRunner', max_iters=40)
checkpoint_config = dict(
    by_epoch=False,
    interval=40,
    meta=dict(
        CLASSES=('building', 'background'), PALETTE=[[255, 0, 0], [0, 0, 0]]))
evaluation = dict(interval=40, metric='mIoU', pre_eval=True)
img_scale = (256, 256)
work_dir = './work_dirs/20220607'
seed = 0
gpu_ids = range(0, 1)

**Comment by LansonChen** [comment]
my annotation files used to be 3-channel tiff data,I write a script to convert them into 1-channel png data.And I  am sure that  the label for bg is 0，for building is 1.

**Comment by LansonChen** [comment]
Does this have anything to do with the network I use？In my view, we could use every net by modifying the num_class.Is it right?

**Comment by LansonChen** [comment]
Does this have anything to do with the network I use？In my view, we could use every net by modifying the num_class.Is it right?

**Comment by LansonChen** [comment]
Here r my configs
![image](https://user-images.githubusercontent.com/33587434/174210987-0f8856ec-e2b7-4116-96c2-da9e8c1d4ca7.png)

![image](https://user-images.githubusercontent.com/33587434/174210473-fd9f2dcc-3273-495a-9469-da52467963e9.png)

**Comment by LansonChen** [comment]
U might notice that I already change the num_classes=2,  reduce_zero_label=False in the config. I don't know what else I can do.

**Comment by MengzhangLI** [comment]
Yes, it is wired. I think it is caused by wrong calculation of `label 1` because it has zero value. `num_classes=2` and  `reduce_zero_label=False` should make label 1 be loaded normally.

Could you attach whole training log and example of image & annotation?

**Comment by LansonChen** [comment]
Hi,apologize for my late reply.
![image](https://user-images.githubusercontent.com/33587434/174436407-4137e2dc-9562-4bf6-a159-8ce57bb62fcc.png)
As far as I know, I found no logs in the work_dir.
Here're my examples:
![image](https://user-images.githubusercontent.com/33587434/174436876-76857524-799e-41fb-9e3e-efe9b3bd5633.png)
![22678915_15](https://user-images.githubusercontent.com/33587434/174436893-f4f1ba03-7c5d-4770-9de3-911d3d4f0554.png)

**Comment by LansonChen** [comment]
training images r tiff format,I took a screenshot of one sample(shape:1500*1500*3), while the annotations r png format(shape:1500*1500).

**Comment by stevie1023** [comment]
> training images r tiff format,I took a screenshot of one sample(shape:1500_1500_3), while the annotations r png format(shape:1500*1500).

Hi, have you figured out the problem?

**Comment by Rinatum** [comment]
same problem

**Comment by stevie1023** [comment]
> same problem

In my case, I need to ensure that the class labels are within [0,num_class-1], which means if I have background and one target, the labels should be [0,1]. Also when loading annotations, check if you need to /255 to make sure the mask values within [0,1](in my case) to correspond to label values.
